common/djangoapps/course_modes/views.py

```diff
@@ -123,15 +126,21 @@ def get(self, request, course_id, error=None):
             "nav_hidden": True,
         }
         if "verified" in modes:
+            verified_mode = modes["verified"]
             context["suggested_prices"] = [
                 decimal.Decimal(x.strip())
-                for x in modes["verified"].suggested_prices.split(",")
+                for x in verified_mode.suggested_prices.split(",")
                 if x.strip()
             ]
-            context["currency"] = modes["verified"].currency.upper()
-            context["min_price"] = modes["verified"].min_price
-            context["verified_name"] = modes["verified"].name
-            context["verified_description"] = modes["verified"].description
+            context["currency"] = verified_mode.currency.upper()
+            context["min_price"] = verified_mode.min_price
+            context["verified_name"] = verified_mode.name
+            context["verified_description"] = verified_mode.description
+
+            if verified_mode.sku:
+                context["use_ecommerce_payment_flow"] = waffle.flag_is_active(request, 'use-ecommerce-payment-flow')
+                context["ecommerce_payment_page"] = urljoin(settings.ECOMMERCE_PUBLIC_URL_ROOT, settings.ECOMMERCE_COURSE_CHECKOUT_PATH)
```

Comment:
The suggested use of a single URL breaks down when performing non-GET operations. I somewhat see the benefits of a cleaner URL; but, the external services would still need to know the correct action, so we aren't exactly gaining anything by moving everything to a single URL.

*The following is not a better solution; but, I'm leaving it here as a reminder to ponder about it further and either make a proposal or eliminate the idea.*
I'd like to see something more akin to a discovery service. The example with which I am most familiar is Elasticsearch. Configuration can be pretty simple: set all your nodes to use the same cluster name. ES will do the rest of the work in terms of moving data around and dealing with primaries and secondaries.

We could take a slightly more manual approach. Each service has an endpoint, `/discover/` that returns a JSON object. That object would include the list of capabilities the service offers, and endpoints. (This is just me riffing, so the schema below may not be ideal.)

```json
GET /discover/

{  
   "capabilities":{  
      "api":{  
         "url":"https://ecommerce.edx.org/api/v2/"
      },
      "single_item_checkout":{  
         "url":"https://ecommerce.edx.org/checkout/single-item/"
      }
   }
}
```